### PR TITLE
Don't exclude imported files in CSS source path

### PR DESCRIPTION
**Pull request by ataylorme**
## Description

Addresses issue #363 by removing the exclude for currently imported CSS files (`reset.css`, `custom-properties.css`, and `custom-media.css`) in the CSS source path. The excluded files have been moved to a configuration option and are added during the styles task, not watch tasks.

The excluded files are not processed and minified as production files, which was the original goal of having them excluded. However, I prefer them in config rather than being hard-coded.

Config options for `customProperties`, `customMedia` and `preserve` have also been added. They directly effect the options passed to postCSS without hard-coding file paths or requiring users to edit gulp files.

## List of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] This pull request relates to a ticket.
- [x] This code is tested.
- [ ] This change has been added to CHANGELOG.md
- [x] I want my code added to WP Rig.

### CODE REVIEW

**Comment by mor10** [comment]
@bamadesigner and @felixarntz had strong opinions on this.

**Comment by benoitchantre** [comment]
Works for me. Thanks @ataylorme.

**Comment by ataylorme** [comment]
@mor10 I updated the PR code and the description. The files are back to being excluded but are now defined in config, rather than being hard-coded in the gulp files.

**Comment by mor10** [comment]
With these new updates, combined with the `editor-styles.css` fix in #367, and with `preserve: true` reinstated, I am happy with this.

Custom properties and media should be passed as-is through to the final code unless the user explicitly turns this feature off. The fallbacks provided by PostCSS are sufficient, and with the fix in #367 work across the front- and back-end.

File: config/config.default.json [hunk 1]
```diff
@@ -12,6 +12,23 @@
       "bypassPort": "8181",
       "https": false
     },
+    "styles": {
+      "preserve": false,
```
Comment (mor10) [inline config/config.default.json]: Set to `true`.

**Comment by benoitchantre** [comment]
> The fallbacks provided by PostCSS are sufficient, and with the fix in #367 work across the front- and back-end.

I have merged this branch locally into #332 (local version) and set `preserve` to `true`, but it doesn't work in the editor.

**Comment by ataylorme** [comment]
@benoitchantre did you merge #367 locally as well? I think that PR is needed too in order to fix the editor.

In either case, I would like to merge this to fix the broken watch task in `master`. If we need to flip the default config value for `preserve` again we can do that in a follow up pull request.

**Comment by benoitchantre** [comment]
> @benoitchantre did you merge #367 locally as well?

Yes. This PR only adds `@import "../custom-media.css";` which is not (yet) required.
`custom-properties.css` was already imported with the latest changes.

Here's what I have when I open the editor.

With `preserve` set to `false`:
capture d ecran 2019-03-06 a 20 55 48

With `preserve` set to `true`:
capture d ecran 2019-03-06 a 20 55 12

**Comment by benoitchantre** [comment]
To get the right preview in the editor with `preserve` set to `true`, `:root` needs to be renamed to `html`.

**Comment by ataylorme** [comment]
> In either case, I would like to merge this to fix the broken watch task in `master`.

Moving ahead with merging.

**Pull request merged**